packages/shell-api/src/replica-set.spec.ts

```diff
@@ -483,4 +484,157 @@ describe('ReplicaSet', () => {
       });
     });
   });
+
+  describe.skip('integration', () => {
+    const port0 = 27017;
+    const host = '127.0.0.1';
+    const replId = 'rs0';
+    const connectionString = `mongodb://${host}:${port0}`; // startTestServer();
+    const cfg = {
+      _id: replId,
+      members: [
+        { _id: 0, host: `${host}:${port0}`, priority: 1 },
+        { _id: 1, host: `${host}:${port0 + 1}`, priority: 0 },
+        { _id: 2, host: `${host}:${port0 + 2}`, priority: 0 }
+      ]
+    };
+    let serviceProvider: CliServiceProvider;
+    let internalState;
+    let mongo;
+    let rs;
+
+    const ensureMaster = (done, timeout): void => {
+      if (timeout > 8000) expect.fail(`Waited for ${host}:${port0} to become master, never happened`);
+      return rs.isMaster().then((res) => {
+        if (res.ismaster) {
+          done();
+        } else { // try again but wait double
+          return setTimeout(async() => {
+            if ((await rs.isMaster()).ismaster) {
+              expect((await rs.conf()).members.length).to.equal(3);
+              done();
+            } else {
+              ensureMaster(done, timeout * 2);
+            }
+          }, timeout);
+        }
+      });
+    };
```

Comment:
This might honestly be a bit easier to understand as an `async` function… something like this?

```js
    const ensureMaster = async(timeout): void => {
      while (!(await rs.isMaster()).ismaster) {
        if (timeout > 8000) 
          return expect.fail(`Waited for ${host}:${port0} to become master, never happened`);
        await delay(timeout);  // delay = util.promisify(setTimeout)
        timeout *= 2; // try again but wait double
      }
      expect((await rs.conf()).members.length).to.equal(3);
    };
```

Reply:
Will do!